### ISSUE TITLE
If still use padding, do mask on several things

### ISSUE THREAD

**Issue by yuhsianglin** [ShuxinLin/nn4nlp_project#19]
If use padding, mask the following:
1. objective function: don't include cross entropy from pair of matching to input padding.
2. attention: when computing alpha_{ij} (doing softmax), don't include padding. (softmax only over non-padding part.)
